plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx

```diff
@@ -19,61 +19,124 @@ const DatasetDetails: React.FC<Props> = ({
   contentSection: ContentSection,
   onDatasetAdd,
 }) => {
+  const [published, setAsPublished] = useState(false);
+
+  const handlePublish = useCallback(() => {
+    // TODO: implement me
+    setAsPublished(true);
+  }, [setAsPublished]);
+
   const handleDatasetAdd = useCallback(() => {
     if (!dataset) return;
     onDatasetAdd(dataset);
   }, [dataset, onDatasetAdd]);
 
   return (
-    <Wrapper>
-      <Title>{dataset.type === "item" && (dataset.cityName ?? dataset.name)}</Title>
-      <ButtonWrapper>
-        <Button disabled={addDisabled} onClick={handleDatasetAdd}>
-          {!addDisabled && <Icon icon="plusCircle" />}
-          {addDisabled ? "シーンに追加済み" : "シーンに追加"}
-        </Button>
-        <Button isShareable={isShareable}>
-          <Icon icon="share" />
-          シェア
-        </Button>
-      </ButtonWrapper>
-      {ContentSection && <ContentSection />}
-    </Wrapper>
+    <>
+      <TopWrapper>
+        <HeaderWrapper>
+          <Title>{dataset.type === "item" && (dataset.cityName ?? dataset.name)}</Title>
+          <PublishButton onClick={handlePublish} published={published} isShareable={isShareable}>
+            <HoverText published={published}>Publish</HoverText>
+            <Text published={published}>{published ? "Published" : "Unpublished"}</Text>
+          </PublishButton>
+        </HeaderWrapper>
+        <ButtonWrapper>
+          <AddButton disabled={addDisabled} onClick={handleDatasetAdd}>
+            {!addDisabled && <Icon icon="plusCircle" />}
+            {addDisabled ? "シーンに追加済み" : "シーンに追加"}
+          </AddButton>
+          <ShareButton isShareable={isShareable}>
+            <Icon icon="share" />
+            シェア
+          </ShareButton>
+        </ButtonWrapper>
+      </TopWrapper>
+      <Wrapper>{ContentSection && <ContentSection />}</Wrapper>
```

Comment:
Unless you need the Wrapper for the structure of the details, can you rewrite like:
```
{ContentSection && (
   <Wrapper>
     <ContentSection />
  </Wrapper>
  )
}
```

Reply:
Done!